### DISCUSSION TITLE
What is different about zksync networks?

### DISCUSSION THREAD

**Discussion by YusufiMekias** [Cyfrin/foundry-full-course-cu#2207] [Q&A]
I am on foundry fund me section lesson 13-14.  I tried to add some extra rpc urls and edit the helperConfig file as needed to support all rpc urls that i added. I added both zksync mainnet and zksync sepolia as the extra urls.  When i run forge test $ANY_ZKSYNC_URL half of my tests fail --> (testing the price feed data is coming in, testing the price feed version is accurate, testing whether i can update the funded data structure). my question what is different about layer 2 networks? all my tests pass when i test on layer 1 networks and using anvil. could someone give me a basic rundown on how zksync works and why its different to deal with than a layer 1 network? PS: this may sound very stupid to an experienced blockchain developer

**Comment by EngrPips** [comment]
Hello @YusufiMekias, You didn't ask a stupid question, and no question is stupid. The issue is working with `ZkSync` using foundry isn't really a smooth process right now, and natively `ZkSync` is designed a slightly different manner than the `Ethereum` blockchain, So somethings might work on Ethereum and not work on `ZkSync` and some things might work on `ZkSync` and not work on `Ethereum` for example `ZkSync` supports native `Account Abstraction` while `Ethereum` have a whole `EIP` (Ethereum Improvement Proposal)` just to achieve `Account Abstraction` functionality. I know most of what I am saying might not really make sense to you so I will advise you to keep going forward with the course and when you reach the `Account Abstraction` part of the course you will understand better as Patrick did better explanation in that section of the course. Here is the link https://docs.zksync.io/build to the ZkSync official documententation incase you are interested.

**Reply by YusufiMekias** [reply]
Thank you for your reply. Although I don't know what account abstraction is yet, I appreciate the reply. I thought my code was broken. Thanks!

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it, and you will know `Account Abstraction` when you get to the section that teaches `Account Abstraction` in the course.
